### PR TITLE
Fixes #3564  Merge profile progress header xml

### LINKED ISSUE #3564: Merge profile_progress_header.xml into single xml file
Currently there are 4 versions of `profile_progress_header.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by bkaur-bkj**
## Explanation

Fixes #3564  
merged portrait and landscape files of tab layout, could not merge the mobile layout files as they had some constraints and other differences too.
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
mobile landscape before 
WhatsApp Image 2021-10-07 at 1 15 42 PM (1)

mobile landscape after changes
WhatsApp Image 2021-10-07 at 1 15 42 PM

WhatsApp Image 2021-10-07 at 1 25 32 PM (1)

WhatsApp Image 2021-10-07 at 1 25 32 PM

After changes tablet portrait

Screenshot_1633716487

After changes tablet landscape

Screenshot_1633716474

### CODE REVIEW

**Comment by bkaur-bkj** [comment]
@rt4914  PTAL, as discussed I made a PR merging the 2 tab layout files but it says some conflicts I tried viewing them but got no idea how to resolve them and not understood the issue.

**Comment by bkaur-bkj** [comment]
@rt4914, I resolved the conflicts but one test is failing

**Comment by rt4914** [comment]
@bkaur-bkj As mentioned in the issue description please  add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

**Comment by rt4914** [comment]
> @rt4914, I resolved the conflicts but one test is failing

@bkaur-bkj The test which is failing is an optional one and also it mostly is failing on develop too so don't worry about it in this PR.

**Comment by bkaur-bkj** [comment]
@rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too.
pls let me know what change should I do

**Comment by rt4914** [comment]
> @rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too. pls let me know what change should I do

@bkaur-bkj For tablet follow these steps:
1. Uninstall the app
2. `Build` -> `Rebuild` the project.
3. Now install the app.
Basically whenever we remove any xml file for tablet we need to uninstall the previous version and install the fresh build.

**Comment by rt4914** [comment]
@bkaur-bkj Also please merge with latest develop as there are conflicts currently.

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj Also please merge with latest develop as there are conflicts currently.

Yes sir Done

**Comment by bkaur-bkj** [comment]
> > @rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too. pls let me know what change should I do
> 
> @bkaur-bkj For tablet follow these steps:
> 
> 1. Uninstall the app
> 2. `Build` -> `Rebuild` the project.
> 3. Now install the app.
>    Basically whenever we remove any xml file for tablet we need to uninstall the previous version and install the fresh build.

yes done worked now

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 1]
```diff
@@ -348,6 +348,13 @@
   <!-- Topic Lessons Fragment -->
   <dimen name="topic_lessons_fragment_layout_padding_bottom">100dp</dimen>
 
+
+  <!-- Profile Progress Header-->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Space before and after the text
`<!-- Profile Progress Header -->`

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 2]
```diff
@@ -350,6 +350,12 @@
   <!-- Topic Lessons Fragment -->
   <dimen name="topic_lessons_fragment_layout_padding_bottom">100dp</dimen>
 
+  <!-- Profile Progress Header-->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Ditto

File: app/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -521,6 +521,12 @@
   <dimen name="story_header_view_layout_margin_start">32dp</dimen>
   <dimen name="story_header_view_layout_margin_end">32dp</dimen>
 
+  <!-- Profile Progress Header-->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Ditto

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -521,6 +521,12 @@
...
+  <dimen name="profile_edit_image_layout_margin_start">42dp</dimen>
+  <dimen name="profile_name_textview_layout_margin_end">202dp</dimen>
+  <dimen name="recently_played_stories_textView_layout_margin_start">42dp</dimen>
+  <dimen name="view_all_textview_margin_end">24dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: These values should be `0dp`  and also add a comment on top mentioning this:
`These values are getting used only in tablet mode and therefore mentioned as 0dp here.`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bkaur-bkj PTAL

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (bkaur-bkj) [reply app/src/main/res/values/dimens.xml]: done

File: app/src/main/res/values/dimens.xml (hunk 3 above)
Comment (bkaur-bkj) [reply app/src/main/res/values/dimens.xml]: done

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 2 above)
Comment (bkaur-bkj) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: done

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 1 above)
Comment (bkaur-bkj) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: done

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -521,6 +521,13 @@
   <dimen name="story_header_view_layout_margin_start">32dp</dimen>
   <dimen name="story_header_view_layout_margin_end">32dp</dimen>
 
+  <!-- These values are getting used only in tablet mode and therefore mentioned as 0dp here -->
+  <!-- Profile Progress Header -->
+  <dimen name="profile_edit_image_layout_margin_start">0dp</dimen>
+  <dimen name="profile_name_textview_layout_margin_end">0dp</dimen>
+  <dimen name="recently_played_stories_textView_layout_margin_start">0dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: There is inconsistency in writing the word `TextView`. You have written it as `textView` as well as `textview`. 

Please use following `text_view`.

This change will affect almost all files.

**Review by rt4914 — CHANGES_REQUESTED** [review]
One comment left.
@bkaur-bkj PTAL

File: app/src/main/res/values/dimens.xml (hunk 5 above)
Comment (bkaur-bkj) [reply app/src/main/res/values/dimens.xml]: okay sir

**Comment by bkaur-bkj** [comment]
@rt4914 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**